sockeye/arguments.py

```diff
@@ -148,11 +148,41 @@ def add_model_parameters(params):
                               type=int_greater_or_equal(0),
                               default=50000,
                               help='Maximum vocabulary size. Default: %(default)s.')
+    model_params.add_argument('--num-words-source',
+                              type=int_greater_or_equal(0),
+                              default=None,
+                              help='Maximum source vocabulary size. Overrides --num-words. Default: %(default)s')
+    model_params.add_argument('--num-words-target',
+                              type=int_greater_or_equal(0),
+                              default=None,
+                              help='Maximum target vocabulary size. Overrides --num-words. Default: %(default)s')
     model_params.add_argument('--word-min-count',
                               type=int_greater_or_equal(1),
                               default=1,
                               help='Minimum frequency of words to be included in vocabularies. Default: %(default)s.')
 
+    model_params.add_argument('--char-seq-encoder',
```

Comment:
1) would it make sense to rename this to a ConvolutionalEncoder? It is only a char-seq encoder if the data fed to it are sequences of characters not words. It could just as well be used for words.
2) We could change the cmdline similarly to the transformer branch where we add an --encoder flag that enables selection of different encoder types (stacked-rnn, transformer, convolutionial)